docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md

```diff
@@ -169,11 +169,24 @@ outside of the existing module.
   if err != nil {
     return err
   }
-  if err := util.WatchOnClusterPaused(controller, clusterToObjectFunc); err != nil {
+  if err := util.WatchOnClusterPaused(controller, clusterToObjectFunc, <logger>); err != nil {
     return err
   }
   ```
   NB: You need to have `cluster.x-k8s.io/cluster-name` applied to all your objects for the mapper to function.
+- In some cases, you'll want to not just watch on Cluster.Spec.Paused changes, but also on
+  Cluster.Status.InfrastructureReady. For those cases `util.WatchOnClusterPausedAndInfrastructureReady` should be used
+  instead. 
+  ```go
+  // Add a watch on clusterv1.Cluster object for paused and infrastructureReady notifications.
+  clusterToObjectFunc, err := util.ClusterToObjectsMapper(mgr.GetClient(), <List object here>, mgr.GetScheme())
+  if err != nil {
+    return err
+  }
+  if err := util.WatchOnClusterPausedAndInfrastructureReady(controller, clusterToObjectFunc, <logger>); err != nil {
+    return err
+  }
```

Comment:
Not sure thrilled about the function signature, thinking about it a little more and brainstorming together: could the extra predicates be passed in to `WatchOnClusterPaused`?  Should `WatchOnClusterPaused` take an Input struct instead of positional arguments like today?

Reply:
Might also be worth to move the predicates to a different package under util, and maybe keep the watch functions here

Reply:
I'm not overly happy with the signature/name here either, however stacking predicates changes the semantics of what we are trying to accomplish.

Will reach out tomorrow to try and see if we can figure something out.

Reply:
Did you reach consensus on this?

Reply:
We're meeting today, you should join us if you have thoughts

Reply:
cool, ping me when that happens.